neutron.cpp

```diff
@@ -0,0 +1,172 @@
+/* 
+ @file      neutron.cpp
+ @brief     contains functions for the Neutron class
+ @author    Luke Eure
+ @date      January 8 2016
+*/
+
+#include "neutron.h"
+
+/*
+ @brief constructor for Neutron class
+ @param position a vector containing the starting position of the neutron
+ @param theta the polar angle of the neutron's initial direction
+ @param phi the azimuthal angle of the neutron's initial direction
+*/
+Neutron::Neutron(std::vector <double> &position, double theta, double phi) {
+    _xyz = position;
+    _neutron_alive = true;
+    _neutron_direction.resize(3);
+    setDirection(theta, phi);
+}
+
+/*
+ @brief deconstructor for Neutron class
+*/
+Neutron::~Neutron() {}
+
+/*
+ @brief tells if the neutron is alive
+ @return a bool (true=neutron alive. false=neutron dead)
+*/
+bool Neutron::alive(){
+    return _neutron_alive;
+}
+
+/*
+ @brief returns the energy group of the neutron
+ @return an int: the energy group of the neutron
+*/
+int Neutron::getGroup() {
+    return _neutron_group;
+}
+
+/*
+ @brief moves the neutron a given distance
+ @param distance the distance the neutron should be moved
+*/
+void Neutron::move(double distance) {
+    for (int i=0; i<3; ++i) {
+        _xyz[i] += _neutron_direction[i]*distance;
+    }
+}
+
+/*
+ @brief moves the neutron a given distance
+ @param distance the distance the neutron should be moved
+*/
+void Neutron::reflect(int axis) {
+    _neutron_direction[axis] *= -1;
+}
+
+/*
+ @brief changes the neutron's cell
+ @param axis the axis along which the cell should be changed
+ @param side whether the cell should be increased or decreased
+*/
+void Neutron::changeCell(int axis, int side) {
+    if (side == 0) {
+        _neutron_cell[axis] -= 1;
+    }
+    if (side == 1) {
+        _neutron_cell[axis] += 1;
+    }
```

Comment:
Strictly speaking, this is fine. But ....
1)  The curly braces are not needed for one line conditionals
2) You check ``side`` twice. While a good compiler might optimize this out, it's good to use an ``else if``
3) Here ``side`` is an ``int`` that might be confusing to some reading it. I would prefer you use the ``min_max`` enum.
4) I believe ``side`` should only take two valid values

Altogether I propose:
```
void Neutron::changeCell(int axis, min_max side) {
    if (side == MIN)
         _neutron_cell[axis]--;
    else
        _neutron_cell[axis]++;
}
```